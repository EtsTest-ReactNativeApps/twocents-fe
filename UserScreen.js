import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

export class UserScreen extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>User Info Screen</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2C2540',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: '#EE933F',
    fontSize: 30,
  }
});