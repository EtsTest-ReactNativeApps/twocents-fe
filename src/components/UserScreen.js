import React, { Component } from 'react';
import { Text, View, StyleSheet, Image, TouchableOpacity, Button } from 'react-native';

export class UserScreen extends Component {

  static navigationOptions = ({ navigation }) => {
      return {
        title: 'My Profile',
        headerStyle: {
          backgroundColor: 'white',
        },
      };
    };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.imageBox}>
          <Image style={styles.image}/>
        </View>
        <View style={styles.info}>
          <View style={styles.textBox}>
            <Text style={styles.mainText}>Username</Text>
            <Text style={styles.minorText}>name from global store</Text>
          </View>
          <View style={styles.textBox}>
            <Text style={styles.mainText}>Email</Text>
            <Text style={styles.minorText}>Email from global store</Text>
          </View>
          <View style={styles.textBox}>
            <Text style={styles.mainText}>Recommendations</Text>
            <Text style={styles.minorText}>from global store</Text>
          </View>
        </View>
        <TouchableOpacity style={styles.button} activeOpacity={.5} >
          <Text style={styles.buttonText}>Log Out</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: '15%',
    flex: 1,
    backgroundColor: '#2C2540',
    alignItems: 'center',
  },
  info: {
    paddingTop: '10%',
    height: '75%',
    width: '100%'
  },
  textBox: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    width: '80%',
    margin: 30,
  },
  mainText: {
    color: '#EE933F',
    fontSize: 30,
  },
  minorText: {
    color: '#CCC0DD',
    fontSize: 25,
    paddingLeft: 30,
  },
  buttonText: {
    color: '#CCC0DD',
    fontSize: 25,
  },
  button: {
    backgroundColor: 'rgba(204, 192, 221, 0.14)',
    borderWidth: 1,
    borderColor: '#EE933F',
    borderRadius: 10,
    width: '80%',
    alignItems: 'center',
    shadowOffset:{  width: 5,  height: 10,  },
    shadowColor: 'black',
    shadowOpacity: 1.0,
  },
  image: {
    height: 100,
    width: 100,
    borderWidth: 1,
    borderColor: 'red'
  },
  imageBox: {
    width: '80%',
    height: '20%',
    borderWidth: 1,
    borderColor: 'green'
  }
});

export default UserScreen;