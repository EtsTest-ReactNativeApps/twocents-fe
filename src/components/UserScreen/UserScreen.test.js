import React from 'react';
import { shallow } from 'enzyme';
import { UserScreen } from './UserScreen';

describe('UserScreen', () => {

  it('should match to snapshot', () => {
    const component = shallow(<UserScreen/>)
    expect(component).toMatchSnapshot()
  });
  
});