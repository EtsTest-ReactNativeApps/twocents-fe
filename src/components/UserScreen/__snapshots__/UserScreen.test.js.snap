// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserScreen should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2C2540",
      "flex": 1,
      "paddingTop": "15%",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "green",
        "borderWidth": 1,
        "height": "20%",
        "width": "80%",
      }
    }
  >
    <Image
      style={
        Object {
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": "75%",
        "paddingTop": "10%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "justifyContent": "flex-start",
          "margin": 30,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EE933F",
            "fontSize": 30,
          }
        }
      >
        Username
      </Text>
      <Text
        style={
          Object {
            "color": "#CCC0DD",
            "fontSize": 25,
            "paddingLeft": 30,
          }
        }
      >
        name from global store
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "justifyContent": "flex-start",
          "margin": 30,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EE933F",
            "fontSize": 30,
          }
        }
      >
        Email
      </Text>
      <Text
        style={
          Object {
            "color": "#CCC0DD",
            "fontSize": 25,
            "paddingLeft": 30,
          }
        }
      >
        Email from global store
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "justifyContent": "flex-start",
          "margin": 30,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EE933F",
            "fontSize": 30,
          }
        }
      >
        Recommendations
      </Text>
      <Text
        style={
          Object {
            "color": "#CCC0DD",
            "fontSize": 25,
            "paddingLeft": 30,
          }
        }
      >
        from global store
      </Text>
    </View>
  </View>
  <TouchableOpacity
    activeOpacity={0.5}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(204, 192, 221, 0.14)",
        "borderColor": "#EE933F",
        "borderRadius": 10,
        "borderWidth": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 10,
          "width": 5,
        },
        "shadowOpacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#CCC0DD",
          "fontSize": 25,
        }
      }
    >
      Log Out
    </Text>
  </TouchableOpacity>
</View>
`;
